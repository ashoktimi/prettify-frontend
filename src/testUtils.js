import React from "react";
import UserContext from "./auth/userContext";


const demoUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  photo_url: null,
};

const UserProvider =
    ({ children, currentUser = demoUser, handleToggleOffcanvas = () => false }) => (
    <UserContext.Provider value={{ currentUser, handleToggleOffcanvas }}>
      {children}
    </UserContext.Provider>
);

function setup(){
  window.matchMedia = jest.fn().mockImplementation(query => {
    return {
      matches: false,
      media: query,
      onchange: null,
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addEventListener: jest.fn(),
      removeEventListener: jest.fn(),
      dispatchEvent: jest.fn(),
    };
  });
}

function teardown(){
   window.matchMedia.mockClear();
   delete window.matchMedia;
}

function commonBeforeEach() {
  setup();
}

function commonAfterEach() {
  teardown();
}

export { UserProvider, commonBeforeEach, commonAfterEach };