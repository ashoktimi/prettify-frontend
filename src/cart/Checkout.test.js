import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import { Provider } from "react-redux";
import { commonBeforeEach, commonAfterEach } from "../testUtils";
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from './cartSlice';
import CheckoutPage from "./Checkout";


beforeEach(commonBeforeEach)
afterEach(commonAfterEach)

describe('CheckoutPage', () => {
 const store = configureStore({
   reducer: {
     cart: cartReducer
   }
 });

 test("renders without crashing", function () {
    render(
      <MemoryRouter>
        <Provider store={store}>
          <CheckoutPage />
        </Provider>    
      </MemoryRouter>   
     );
  });
  
  test("matches snapshot", function () {
    const { asFragment } = 
     render(
      <MemoryRouter>
        <Provider store={store}>
          <CheckoutPage />       
        </Provider>    
      </MemoryRouter>  
     );
    expect(asFragment()).toMatchSnapshot();
  });

})