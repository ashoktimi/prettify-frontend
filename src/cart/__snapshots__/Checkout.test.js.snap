// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckoutPage matches snapshot 1`] = `
<DocumentFragment>
  <div
    style="width: 70%; margin-left: 15%; margin-top: 5rem; display: flex; justify-content: space-between;"
  >
    <form
      class=""
    >
      <h2>
        Personal Information
      </h2>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formFirstName"
            >
              First Name:
            </label>
            <input
              class="form-control"
              id="formFirstName"
              name="firstName"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formLastName"
            >
              Last Name:
            </label>
            <input
              class="form-control"
              id="formLastName"
              name="lastName"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="formEmail"
        >
          Email:
        </label>
        <input
          class="form-control"
          id="formEmail"
          name="email"
          type="email"
          value=""
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formCountry"
            >
              Country:
            </label>
            <input
              class="form-control"
              id="formCountry"
              name="country"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formPostalCode"
            >
              Postal Code:
            </label>
            <input
              class="form-control"
              id="formPostalCode"
              name="postalCode"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div>
        <label
          class="form-label"
          for="formPhone"
        >
          Phone Number:
        </label>
        <input
          class="form-control"
          id="formPhone"
          name="phone"
          type="tel"
          value=""
        />
      </div>
      <h2>
        Payment Details
      </h2>
      <div>
        <label
          class="form-label"
          for="formCardNumber"
        >
          Card Number:
        </label>
        <input
          class="form-control"
          id="formCardNumber"
          name="cardNumber"
          type="text"
          value=""
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formExpMonth"
            >
              Expiration Month:
            </label>
            <input
              class="form-control"
              id="formExpMonth"
              name="expMonth"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formExpYear"
            >
              Expiration Year:
            </label>
            <input
              class="form-control"
              id="formExpYear"
              name="expYear"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="formCVV"
            >
              CVV:
            </label>
            <input
              class="form-control"
              id="formCVV"
              name="cvv"
              type="text"
              value=""
            />
          </div>
        </div>
        <div
          class="col"
        >
          <div>
            <label
              class="form-label"
              for="billingAddress"
            >
              Billing Address:
            </label>
            <input
              class="form-control"
              id="billingAddress"
              name="billing"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <button
        class="btn btn-warning"
        style="margin-top: 10px;"
        type="button"
      >
        Submit
      </button>
    </form>
    <div
      style="background-color: white; padding: 0px 3rem 5rem 3rem; width: 40%; margin-left: 1rem; margin-top: 5rem;"
    >
      <h5
        style="text-align: center; margin-bottom: 1.5rem; margin-top: 2rem;"
      >
        IN YOUR BAG
      </h5>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Subtotal:
        </p>
        <p>
          $0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Estimated Shipping:
        </p>
        <p>
          0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <p>
          Estimated Tax:
        </p>
        <p>
          0
        </p>
      </div>
      <div
        style="display: flex; justify-content: space-between;"
      >
        <h5>
          Total:
        </h5>
        <h5>
           $0
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
